src/proxy.rs

```diff
@@ -873,41 +890,115 @@ fn is_cgi() -> bool {
 }
 
 #[cfg(target_os = "windows")]
-fn get_from_registry_impl() -> Result<RegistryProxyValues, Box<dyn Error>> {
+fn get_from_platform_impl() -> Result<Option<String>, Box<dyn Error>> {
     let hkcu = RegKey::predef(HKEY_CURRENT_USER);
     let internet_setting: RegKey =
         hkcu.open_subkey("Software\\Microsoft\\Windows\\CurrentVersion\\Internet Settings")?;
     // ensure the proxy is enable, if the value doesn't exist, an error will returned.
     let proxy_enable: u32 = internet_setting.get_value("ProxyEnable")?;
     let proxy_server: String = internet_setting.get_value("ProxyServer")?;
 
-    Ok((proxy_enable, proxy_server))
+    if proxy_enable == 1 {
+        Some(proxy_server)
+    } else {
+        None
+    }
 }
 
-#[cfg(target_os = "windows")]
-fn get_from_registry() -> Option<RegistryProxyValues> {
-    get_from_registry_impl().ok()
-}
+#[cfg(target_os = "macos")]
+fn parse_setting_from_dynamic_store(
+    proxies_map: &CFDictionary<CFString, CFType>,
+    enabled_key: CFStringRef,
+    host_key: CFStringRef,
+    port_key: CFStringRef,
+    scheme: &str,
+) -> Option<String> {
+    let proxy_enabled = proxies_map
+        .find(enabled_key)
+        .and_then(|flag| flag.downcast::<CFNumber>())
+        .and_then(|flag| flag.to_i32())
+        .unwrap_or(0)
+        == 1;
+
+    if proxy_enabled {
+        let proxy_host = proxies_map
+            .find(host_key)
+            .and_then(|host| host.downcast::<CFString>())
+            .map(|host| host.to_string())
+            .unwrap_or_default();
```

Comment:
If no proxy_host is set should this return `None`? It seems odd we'd be returning a value of `Some("https=")` in that case

Reply:
Agreed - refactored this a bit to only return strings for the host or host+proxy cases.